multicast/hear.py

```diff
@@ -0,0 +1,385 @@
+import sys
+
+
+try:
+	if 'multicast' not in sys.modules:
+		from . import multicast as multicast
+	else:  # pragma: no branch
+		multicast = sys.modules["""multicast"""]
+	_BLANK = multicast._BLANK
+except Exception as importErr:
+	del importErr
+	import multicast as multicast
+
+
+try:
+	import socketserver
+	from socketserver import threading as threading
+	from multicast import argparse as argparse
+	from multicast import unicodedata as unicodedata
+	from multicast import socket as socket
+	from multicast import struct as struct
+	depends = [
+		unicodedata, socket, struct, argparse
+	]
+	for unit in depends:
+		try:
+			if unit.__name__ is None:  # pragma: no branch
+				raise ImportError(
+					str("[CWE-440] module failed to import {}.").format(str(unit))
+				)
+		except Exception:  # pragma: no branch
+			raise ImportError(str("[CWE-758] Module failed completely."))
+except Exception as err:
+	raise ImportError(err)
+
+
+from . import recv as recv
+from . import send as send
+
+
+class McastServer(socketserver.UDPServer):
+
+	def server_activate(self):
+		print(str("server_activate"))
+		self.open_for_request()
+		super(McastServer, self).server_activate()
+
+	def open_for_request(self):
+		print(str("open_request"))
+		old_socket = self.socket
+		(tmp_addr, tmp_prt) = old_socket.getsockname()
+		recv.endSocket(old_socket)
+		self.socket = recv.joinstep([tmp_addr], tmp_prt, None, tmp_addr, recv.genSocket())
+
+	def server_bind(self):
+		print(str("server_bind"))
+		super(McastServer, self).server_bind()
+		print(str("bound on: {}").format(str(self.socket.getsockname())))
+
+	def close_request(self, request):
+		print(str("close_request"))
+		self.open_for_request()
+		super(McastServer, self).close_request(request)
+
+	def handle_error(self, request, client_address):
+		print(str("handle_error"))
+		if request is not None:
+			if request[0] is not None:
+				if """STOP""" in str(request[0]):
+					def kill_func(a_server):
+						if a_server is not None:
+							a_server.shutdown()
+					end_thread = threading.Thread(name="Kill_Thread", target=kill_func, args=[self])
+					end_thread.start()
+		super(McastServer, self).handle_error(request, client_address)
+
+
+class MyUDPHandler(socketserver.BaseRequestHandler):
+	def handle(self):
+		data = self.request[0].strip()
+		socket = self.request[1]
+		print(str("{} wrote:").format(self.client_address[0]))
+		print(data)
+		socket.sendto(data.upper(), self.client_address)
+
+
+class HearUDPHandler(socketserver.BaseRequestHandler):
+	def handle(self):
+		data = self.request[0].strip()
+		socket = self.request[1]
+		print(str("{} SAYS: {} to {}").format(
+			self.client_address[0], str(data), "ALL"
+		))
+		if data is not None:
+			myID = str(socket.getsockname()[0])
+			print(str("{me} HEAR: [{you} SAID {what}]").format(me=myID, you=self.client_address, what=str(data)))
+			print(str("{me} SAYS [ HEAR [ {what} SAID {you} ] from {me} ]").format(me=myID, you=self.client_address, what=str(data)))
```

Comment:
```suggestion
			print(
				str(
					"{me} SAYS [ HEAR [ {what} SAID {you} ] from {me} ]"
				).format(
					me=myID, you=self.client_address, what=str(data)
				)
			)
```
use PEP-8 or refactor into helper-function